pkg/dock/discovery/discovery.go

```diff
@@ -129,9 +131,18 @@ func (pdd *provisionDockDiscoverer) Init() error {
 func (pdd *provisionDockDiscoverer) Discover() error {
 	// Clear existing pool info
 	pdd.pols = pdd.pols[:0]
+	var pols []*model.StoragePoolSpec
+	var err error
 	for _, dck := range pdd.dcks {
 		// Call function of StorageDrivers configured by storage drivers.
-		pols, err := drivers.Init(dck.DriverName).ListPools()
+		if (dck.DriverName == config.NFSDriverType) {
+			pols, err = fd.Init(dck.DriverName).ListPools()
```

Comment:
how to support different file share driver rather than NFSDriverType?

Reply:
Resolved now